src/Structures/Base.js

```diff
@@ -505,14 +505,17 @@ class Base {
     /**
      * Fetches data from the database/API.
      *
+     * @param {options}             Fetch options
+     * @param {options.params}      Query params
+     * @param {options.headers}     Query headers
      * @returns {Promise}
      */
-    fetch() {
+    fetch(options = {}) {
         let config = () => ({
             url:     this.getFetchURL(),
             method:  this.getFetchMethod(),
-            params:  this.getFetchQuery(),
-            headers: this.getFetchHeaders(),
+            params:  _.assign({}, this.getFetchQuery(), options.params),
+            headers: _.assign({}, this.getFetchHeaders(), options.headers),
```

Comment:
I think we could do a `defaultsDeep` here, rather than specific overrides.

Reply:
humm, like this ?

Reply:
But with that, it's possible to override existing parameters if user want to do this. With `defaultsDeep`, it's not possible

Reply:
Something like this (we actually don't need the "deep"):

```
        let config = () => _.defaults(options, {
            url:     this.getFetchURL(),
            method:  this.getFetchMethod(),
            params:  this.getFetchQuery(),
            headers: this.getFetchHeaders(),
        });
```

This allows you to override anything you want.

Reply:
got it